nltk/ccg/api.py

```diff
@@ -5,43 +5,58 @@
 # URL: <http://nltk.org/>
 # For license information, see LICENSE.TXT
 from __future__ import unicode_literals
+from abc import ABCMeta, abstractmethod
+from six import add_metaclass
+
 from nltk.internals import raise_unorderable_types
 from nltk.compat import (total_ordering, python_2_unicode_compatible,
                          unicode_repr)
 
+
+@add_metaclass(ABCMeta)
 @total_ordering
 class AbstractCCGCategory(object):
     '''
     Interface for categories in combinatory grammars.
     '''
 
-    # Returns true if the category is primitive
+    @abstractmethod
     def is_primitive(self):
-        raise NotImplementedError()
+        """
+        Returns true if the category is primitive.
+        """
 
-    # Returns true if the category is a function application
+    @abstractmethod
     def is_function(self):
-        raise NotImplementedError()
+        """
+        Returns true if the category is a function application.
+        """
 
-    # Returns true if the category is a variable
+    @abstractmethod
     def is_var(self):
-        raise NotImplementedError()
+        """
+        Returns true if the category is a variable.
+        """
 
-    # Takes a set of (var, category) substitutions, and replaces every
-    # occurrence of the variable with the corresponding category
-    def substitute(self,substitutions):
-        raise NotImplementedError()
+    @abstractmethod
+    def substitute(self, substitutions):
+        """
+        Takes a set of (var, category) substitutions, and replaces every
+        occurrence of the variable with the corresponding category.
+        """
 
-    # Determines whether two categories can be unified.
-    #  - Returns None if they cannot be unified
-    #  - Returns a list of necessary substitutions if they can.'''
-    def can_unify(self,other):
-        raise NotImplementedError()
+    @abstractmethod
+    def can_unify(self, other):
+        """
+        Determines whether two categories can be unified.
+         - Returns None if they cannot be unified
+         - Returns a list of necessary substitutions if they can.
+        """
 
     # Utility functions: comparison, strings and hashing.
-
+    @abstractmethod
     def __str__(self):
-        raise NotImplementedError()
+        pass
```

Comment:
The `pass` here is inconsistent with the rest of the `@abstractmethod`. Is it necessary?